src/Screens/Explore/DiveSite.tsx

```diff
@@ -248,17 +248,17 @@ const DiveSite: FunctionComponent<DiveSiteProps> = ({ navigation, route }) => {
       <ScrollView
         style={styles.scrollContainer}
         showsVerticalScrollIndicator={false}>
-        <SharedElement id={`item.${navObjectSpot.id}.image`}>
+        <SharedElement id={`item.${(navObjectSpot as Spot).id}.image`}>
```

Comment:
rather than re-casting this here all the time, we should re-cast it at the top of the file when it's initialized

Reply:
doing this is basically just ignoring the type warnings. if navObjectSpot is sometimes optional we need to add logic that ensures that it's defined

Reply:
Sure. Was a last minute change, and I did not think of that.

Reply:
Actually can't typecast when initialised (linter will bleat) because it is an optional navigation param. I can use a question mark for the values that don't need typecasting and typecast the only one that needs it.

Reply:
Not really. If you check line 227, you'd see where I handled the case where navObjectSpot is undefined. Else navigating to DiveSpot from search would break the app.

I deliberately made it optional because I know not all redirects to that page will pass along the navObjectSpot.

Reply:
yeah the compiler is correct that this value isn't guaranteed to exist. so we need to create code patterns that ensure it exists, like adding the question mark or checking for null some other way. otherwise for example if the user enters this screen from a deeplink (where they would have just the beach ID and not the navObjectSpot, then it could result in an error/crash)

Reply:
Yes. Except I purposely told the compiler that that value might not exist. It is reading from a Navigator type object for that page that I defined, and deliberately made optional.

If I change that value and make the compiler thing it will always exist, that error will disappear.

And I've already handled the case for when the object doesn't exist (see line 227 of the same file. mentioned in my previous reply). Deeplink is not the only use case for this. It also happens when the user navigates from explore screen search results. those typically don't carry beach objects with them.

Reply:
you may be right, but ill need to read the full code more closely when i get time later today.

in general having to cast things this way isn't ideal and there is a better way to do it that still allows the compiler to help catch future bugs, which is why i'm skeptical

Reply:
just to elaborate, generally the only times you want to cast are when things are at the time something is coming in from an external interface, and you know it's guaranteed to exist otherwise something else has gone wrong (eg an API response with a specific format).

In this case the data could be coming from an external interface (eg deeplink), but the correct typing is still optional as you stated, since that data isn't guaranteed to exist. So then for the rest of the code we shouldn't be doing typecasting and instead working with the compiler to ensure the values exist.